### ISSUE TITLE
Postfix running into `owned or writable by non-root users`

### ISSUE THREAD

**Issue by macwinnie** [mailcow/mailcow-dockerized#4129]
**Prior to placing the issue, please check following:** *(fill out each checkbox with an `X` once done)*
- [X] I understand that not following or deleting the below instructions will result in immediate closure and/or deletion of my issue.
- [X] I have understood that this bug report is dedicated for bugs, and not for support-related inquiries.
- [X] I have understood that answers are voluntary and community-driven, and not commercial support.
- [X] I have verified that my issue has not been already answered in the past. I also checked previous [issues](https://github.com/mailcow/mailcow-dockerized/issues).

## Summary

After running some time, the working Postfix runs into a non-working state with loads of `owned or writable by non-root users` errors.

I've already found the issue – and the described solution was to run the `docker-compose up -d` as `root` and also to checkout the whole Git repository as `root` user. Both is followed, the services start up and work well for some time but then runs into that state.

To (temporarily) fix the problem, one has to run `docker exec -it mailcowdockerized_postfix-mailcow_1 bash -c 'chmod 0755 /usr/lib/postfix/*.so'` followed by `docker restart mailcowdockerized_postfix-mailcow_1`.

## Logs

The logs start at the point, the send procedure of a message was successfully finished.

```
Jun  6 16:57:29 236acb9ece4c postfix/smtp[407]: 6A68D1421460: to=<***@gmail.com>, relay=gmail-smtp-in.l.google.com[0::0]:25, delay=0.79, delays=0.09/0/0.28/0.42, dsn=2.0.0, status=sent (250 2.0.0 OK  1622991449 o3si13451495lft.532 - gsmtp)
Jun  6 16:57:29 236acb9ece4c postfix/qmgr[399]: 6A68D1421460: removed
Jun  6 17:01:57 236acb9ece4c postfix/qmgr[399]: 78EFB1421448: from=<watchdog@invalid>, size=275, nrcpt=1 (queue active)
Jun  6 17:01:57 236acb9ece4c postfix/qmgr[399]: 78EFB1421448: removed
Jun  6 17:11:57 236acb9ece4c postfix/qmgr[399]: 9409A1421452: from=<watchdog@invalid>, size=275, nrcpt=1 (queue active)
Jun  6 17:11:57 236acb9ece4c postfix/qmgr[399]: 9409A1421452: removed
Jun  6 17:20:53 236acb9ece4c postfix/proxymap[476]: warning: /etc/postfix/dynamicmaps.cf: file is owned or writable by non-root users -- skipping this file
Jun  6 17:20:53 236acb9ece4c postfix/proxymap[476]: error: unsupported dictionary type: mysql
Jun  6 17:20:53 236acb9ece4c postfix/proxymap[476]: warning: mysql:/opt/postfix/conf/sql/mysql_mbr_access_maps.cf is unavailable. unsupported dictionary type: mysql
Jun  6 17:21:03 236acb9ece4c postfix/proxymap[476]: error: unsupported dictionary type: mysql
Jun  6 17:21:03 236acb9ece4c postfix/proxymap[476]: error: unsupported dictionary type: mysql
Jun  6 17:21:03 236acb9ece4c postfix/proxymap[476]: error: unsupported dictionary type: mysql
Jun  6 17:21:03 236acb9ece4c postfix/proxymap[476]: error: unsupported dictionary type: mysql
Jun  6 17:21:03 236acb9ece4c postfix/proxymap[476]: error: unsupported dictionary type: mysql
Jun  6 17:21:03 236acb9ece4c postfix/proxymap[476]: error: unsupported dictionary type: mysql
Jun  6 17:21:03 236acb9ece4c postfix/proxymap[476]: error: unsupported dictionary type: mysql
Jun  6 17:21:03 236acb9ece4c postfix/proxymap[476]: error: unsupported dictionary type: mysql
Jun  6 17:21:03 236acb9ece4c postfix/proxymap[476]: error: unsupported dictionary type: mysql
Jun  6 17:21:03 236acb9ece4c postfix/proxymap[476]: error: unsupported dictionary type: mysql
Jun  6 17:21:03 236acb9ece4c postfix/proxymap[476]: error: unsupported dictionary type: mysql
Jun  6 17:21:59 236acb9ece4c postfix/sogo/smtpd[480]: warning: /etc/postfix/dynamicmaps.cf: file is owned or writable by non-root users -- skipping this file
Jun  6 17:21:59 236acb9ece4c postfix/sogo/smtpd[480]: connect from mailcowdockerized_sogo-mailcow_1.mailcowdockerized_mailcow-network[172.22.1.248]
Jun  6 17:21:59 236acb9ece4c postfix/proxymap[476]: warning: mysql:/opt/postfix/conf/sql/mysql_virtual_alias_maps.cf is unavailable. unsupported dictionary type: mysql
Jun  6 17:21:59 236acb9ece4c postfix/trivial-rewrite[405]: warning: virtual_alias_domains: proxy:mysql:/opt/postfix/conf/sql/mysql_virtual_alias_maps.cf: table lookup problem
Jun  6 17:21:59 236acb9ece4c postfix/trivial-rewrite[405]: warning: virtual_alias_domains lookup failure
Jun  6 17:21:59 236acb9ece4c postfix/proxymap[476]: warning: mysql:/opt/postfix/conf/sql/mysql_virtual_alias_maps.cf is unavailable. unsupported dictionary type: mysql
Jun  6 17:21:59 236acb9ece4c postfix/trivial-rewrite[405]: warning: virtual_alias_domains: proxy:mysql:/opt/postfix/conf/sql/mysql_virtual_alias_maps.cf: table lookup problem
Jun  6 17:21:59 236acb9ece4c postfix/trivial-rewrite[405]: warning: virtual_alias_domains lookup failure
Jun  6 17:21:59 236acb9ece4c postfix/proxymap[476]: warning: mysql:/opt/postfix/conf/sql/mysql_virtual_alias_maps.cf is unavailable. unsupported dictionary type: mysql
Jun  6 17:21:59 236acb9ece4c postfix/trivial-rewrite[405]: warning: virtual_alias_domains: proxy:mysql:/opt/postfix/conf/sql/mysql_virtual_alias_maps.cf: table lookup problem
Jun  6 17:21:59 236acb9ece4c postfix/trivial-rewrite[405]: warning: virtual_alias_domains lookup failure
Jun  6 17:21:59 236acb9ece4c postfix/sogo/smtpd[480]: NOQUEUE: reject: RCPT from mailcowdockerized_sogo-mailcow_1.mailcowdockerized_mailcow-network[172.22.1.248]: 451 4.3.0 <***@***.**>: Temporary lookup failure; from=<***@***.**> to=<***@gmail.com> proto=ESMTP helo=<3b4e5052e647>
Jun  6 17:21:59 236acb9ece4c postfix/sogo/smtpd[480]: lost connection after RCPT from mailcowdockerized_sogo-mailcow_1.mailcowdockerized_mailcow-network[172.22.1.248]
Jun  6 17:21:59 236acb9ece4c postfix/sogo/smtpd[480]: disconnect from mailcowdockerized_sogo-mailcow_1.mailcowdockerized_mailcow-network[172.22.1.248] ehlo=1 auth=1 mail=1 rcpt=0/1 commands=3/4
Jun  6 17:22:16 236acb9ece4c postfix/proxymap[476]: warning: mysql:/opt/postfix/conf/sql/mysql_virtual_alias_maps.cf is unavailable. unsupported dictionary type: mysql
Jun  6 17:22:16 236acb9ece4c postfix/trivial-rewrite[405]: warning: virtual_alias_domains: proxy:mysql:/opt/postfix/conf/sql/mysql_virtual_alias_maps.cf: table lookup problem
Jun  6 17:22:16 236acb9ece4c postfix/trivial-rewrite[405]: warning: virtual_alias_domains lookup failure
Jun  6 17:22:16 236acb9ece4c postfix/trivial-rewrite[405]: warning: virtual_alias_domains: proxy:mysql:/opt/postfix/conf/sql/mysql_virtual_alias_maps.cf: table lookup problem
Jun  6 17:22:16 236acb9ece4c postfix/proxymap[476]: warning: mysql:/opt/postfix/conf/sql/mysql_virtual_alias_maps.cf is unavailable. unsupported dictionary type: mysql
Jun  6 17:22:16 236acb9ece4c postfix/proxymap[476]: error: unsupported dictionary type: mysql
Jun  6 17:22:16 236acb9ece4c postfix/proxymap[476]: fatal: too many errors - program terminated
Jun  6 17:22:17 236acb9ece4c postfix/master[397]: warning: process /usr/lib/postfix/sbin/proxymap pid 476 exit status 1
Jun  6 17:22:18 236acb9ece4c postfix/proxymap[485]: warning: /etc/postfix/dynamicmaps.cf: file is owned or writable by non-root users -- skipping this file
Jun  6 17:22:18 236acb9ece4c postfix/proxymap[485]: error: unsupported dictionary type: mysql
Jun  6 17:22:18 236acb9ece4c postfix/proxymap[485]: warning: mysql:/opt/postfix/conf/sql/mysql_virtual_domains_maps.cf is unavailable. unsupported dictionary type: mysql
Jun  6 17:22:18 236acb9ece4c postfix/trivial-rewrite[405]: warning: virtual_mailbox_domains: proxy:mysql:/opt/postfix/conf/sql/mysql_virtual_domains_maps.cf: table lookup problem
Jun  6 17:22:18 236acb9ece4c postfix/proxymap[485]: error: unsupported dictionary type: mysql
Jun  6 17:22:18 236acb9ece4c postfix/proxymap[485]: warning: mysql:/opt/postfix/conf/sql/mysql_mbr_access_maps.cf is unavailable. unsupported dictionary type: mysql
Jun  6 17:23:17 236acb9ece4c postfix/proxymap[485]: warning: mysql:/opt/postfix/conf/sql/mysql_mbr_access_maps.cf is unavailable. unsupported dictionary type: mysql
```

## Reproduction

Clone the `mailcow/mailcow-dockerized` repo as `root`, run the `./generate_config.sh` as `root` and run `docker-compose up -d` as `root`.

## System information

| Question | Answer |
| --- | --- |
| My operating system | Ubuntu 20.04 |
| Is Apparmor, SELinux or similar active? | no |
| Virtualization technlogy (KVM, VMware, Xen, etc - **LXC and OpenVZ are not supported** | bare metal |
| Server/VM specifications (Memory, CPU Cores) | 8 Cores, Intel Xeon E3-1245, 32 GB RAM |
| Docker Version (`docker version`) | Docker Engine - Community, 20.10.7 |
| Docker-Compose Version (`docker-compose version`) | docker-compose version 1.29.2 |
| Reverse proxy (custom solution) | Træfik |

**Comment by waja** [comment]
Have you checked you are not running more then one (postfix) containers?

**Comment by andryyy** [comment]
Not a bug. Check the docs.

**Comment by macwinnie** [comment]
@andryyy thanks for the hint – I think, reading the docs as lecture the last weeks before setting the mailcow up and running fulfills the task. Probably you could give me a hint, where in the docs I've missed the relevant part?

**Comment by macwinnie** [comment]
> Have you checked you are not running more then one (postfix) containers?

checked that – single Postfix on the host ...

**Comment by macwinnie** [comment]
Hi @andryyy,

a hint, where in the docs, would be really great. I searched through it thoroughly again, but did not find anything.

I still think about the error as a bug, since fixing the file permissions resolves the `4.3.0 <***@***.**>: Temporary lookup failure` error when trying to send an email.

**Comment by macwinnie** [comment]
Found a way to fix that.
